edgelet/iotedge/src/config/apply.rs

```diff
@@ -252,7 +252,90 @@ fn execute_inner(
+                        uid: aziotcs_uid.as_raw(),
+                        keys: vec![edgelet_core::AZIOT_EDGED_CA_ALIAS.to_owned()],
+                    };
+
+                    let mut keys = std::collections::BTreeMap::default();
+
+                    let auth = common_config::apply::set_est_auth(
+                        &auth,
+                        &mut certd_config.preloaded_certs,
+                        &mut keys,
+                        &mut aziotcs_principal,
+                    );
+
+                    keyd_config.principal.push(aziotcs_principal);
+
+                    keyd_config.preloaded_keys.append(
+                        &mut keys
+                            .into_iter()
+                            .map(|(id, location)| (id, location.to_string()))
+                            .collect(),
+                    );
+
+                    certd_config.cert_issuance.certs.insert(
+                        edgelet_core::AZIOT_EDGED_CA_ALIAS.to_owned(),
+                        aziot_certd_config::CertIssuanceOptions {
+                            method: aziot_certd_config::CertIssuanceMethod::Est { url, auth },
+                            common_name,
+                            expiry_days,
+                        },
+                    );
+
+                    (
+                        Some(edgelet_core::AZIOT_EDGED_CA_ALIAS.to_owned()),
+                        Some(edgelet_core::AZIOT_EDGED_CA_ALIAS.to_owned()),
+                    )
+                }
+                common_config::super_config::CertIssuanceMethod::LocalCa => {
+                    certd_config.cert_issuance.certs.insert(
+                        edgelet_core::AZIOT_EDGED_CA_ALIAS.to_owned(),
+                        aziot_certd_config::CertIssuanceOptions {
+                            method: aziot_certd_config::CertIssuanceMethod::LocalCa,
+                            common_name,
+                            expiry_days
+                        }
+                    );
+
+                    keyd_config.principal.push(aziot_keyd_config::Principal {
+                        uid: aziotcs_uid.as_raw(),
+                        keys: vec![edgelet_core::AZIOT_EDGED_CA_ALIAS.to_owned()],
+                    });
+
+                    (
+                        Some(edgelet_core::AZIOT_EDGED_CA_ALIAS.to_owned()),
+                        Some(edgelet_core::AZIOT_EDGED_CA_ALIAS.to_owned()),
+                    )
+                },
+                common_config::super_config::CertIssuanceMethod::SelfSigned => {
+                    // This is equivalent to a quickstart CA.
+                    set_quickstart_ca(
+                        &mut keyd_config,
+                        &mut certd_config,
+                        aziotcs_uid,
+                        expiry_days,
+                        common_name,
+                    );
+
+                    (
+                        Some(edgelet_core::AZIOT_EDGED_CA_ALIAS.to_owned()),
```

Comment:
@arsing Is it correct here to provide the name of the edge CA cert/key here instead of (None, None) as in Quickstart? I wasn't sure about this.

Reply:
Yes.